perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -974,3 +975,214 @@ def MoveFile(self, target, source_path, remote_path=''):
+
+  def JujuSet(self, service, params=[]):
+    """Set the configuration options on a deployed service.
+
+    Args:
+      service: The name of the service.
+      params: A list of key=values pairs.
+    """
+    output, _ = self.RemoteHostCommand(
+        'juju set %s %s' % (service, ' '.join(params)))
+    return output.strip()
+
+  @vm_util.Retry(poll_interval=30, timeout=3600)
+  def JujuWait(self):
+    """Wait for all deployed services to be installed, configured, and idle."""
+    status = yaml.load(self.JujuStatus())
+    for service in status['services']:
+      ss = status['services'][service]['service-status']['current']
+
+      # Accept blocked because the service may be waiting on relation
+      if ss not in ['active', 'unknown']:
+          raise errors.Juju.TimeoutException(
+              'Service %s is not ready; status is %s' % (service, ss))
+
+      if ss in ['error']:
+        # The service has failed to deploy.
+        debuglog = self.JujuRun('juju debug-log --limit 200')
+        logging.warn(debuglog)
+        raise errors.Juju.UnitErrorException(
+            'Service %s is in an error state' % service)
+
+      for unit in status['services'][service]['units']:
+        unit_data = status['services'][service]['units'][unit]
+        ag = unit_data['agent-state']
+        if ag != 'started':
+          raise errors.Juju.TimeoutException(
+              'Service %s is not ready; agent-state is %s' % (service, ag))
+
+        ws = unit_data['workload-status']['current']
+        if ws not in ['active', 'unknown']:
+          raise errors.Juju.TimeoutException(
+              'Service %s is not ready; workload-state is %s' % (service, ws))
+
+  def JujuDeploy(self, charm, vm_group, units=1):
+    """Deploy (and scale) this service to the machines in its vm group.
+
+    Args:
+      charm: The charm to deploy, i.e., cs:trusty/ubuntu.
+      units: Optional number of units to create.
+      vm_group: The name of vm_group the unit(s) should be deployed to.
+    """
+
+    # Find the already-deployed machines belonging to this vm_group
+    machines = []
+    for machine_id, unit in self.machines.iteritems():
+      if unit.vm_group == vm_group:
+        machines.append(machine_id)
+
+    # Deploy the first machine
+    resp, _ = self.RemoteHostCommand(
+        "juju deploy %s --to %s" % (charm, machines.pop()))
+
+    # Get the name of the service
+    service = charm[charm.rindex('/') + 1:]
+
+    # Deploy to the remaining machine(s)
+    for machine in machines:
+      resp, _ = self.RemoteHostCommand(
+          "juju add-unit %s --to %s" % (service, machine))
+
+  def JujuRelate(self, service1, service2):
+    """Create a relation between two services.
+
+    Args:
+      service1: The first service to relate.
+      service2: The second service to relate.
+    """
+    resp, _ = self.RemoteHostCommand(
+        "juju add-relation %s %s" % (service1, service2))
+
+  def Install(self, package_name):
```

Comment:
Is there an equivalent way to uninstall jujus? When running on pre-provisioned machines, we try to clean everything up afterwards. Defining a [PackageCleanup](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_virtual_machine.py#L223) method  and an [Uninstall](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_virtual_machine.py#L783)  method + JujuUninstall functions in any packages where PackageCleanup isn't enough (similar to the AptUninstall functions [here](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/d7de2e2aa82650190b4640f8d64657846608144b/perfkitbenchmarker/linux_packages/pip.py#L57)) should accomplish this.

You could also just add a TODO and take care of this in a subsequent PR since this only affects pre-provisioned machines